pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -118,9 +111,15 @@ public QueryEnvironment(Config config) {
     String database = config.getDatabase();
     PinotCatalog catalog = new PinotCatalog(config.getTableCache(), database);
     CalciteSchema rootSchema = CalciteSchema.createRootSchema(false, false, database, catalog);
+    Properties connectionConfigProperties = new Properties();
+    connectionConfigProperties.setProperty(CalciteConnectionProperty.CASE_SENSITIVE.camelName(), Boolean.toString(
+        config.getTableCache() == null
```

Comment:
Will the table cache ever be `null` here? Doesn't look like it should be.

Reply:
Just defensive programming :-)

As long as `QueryEnvironment.getTableCache()` is annotated as `@Nullable`, I'd rather be safe than sorry.

Reply:
I missed that, but it looks like it can be nullable only in tests which should probably be fixed. But that's outside the scope of this PR and can be done separately.